test/regress_buffer.c

```diff
@@ -2494,28 +2494,43 @@ test_evbuffer_peek(void *info)
 static void
 test_evbuffer_freeze(void *ptr)
 {
-	struct evbuffer *buf = NULL, *tmp_buf=NULL;
+	struct evbuffer *buf = NULL, *buf_two = NULL, *tmp_buf = NULL;
 	const char string[] = /* Year's End, Richard Wilbur */
 	    "I've known the wind by water banks to shake\n"
 	    "The late leaves down, which frozen where they fell\n"
 	    "And held in ice as dancers in a spell\n"
 	    "Fluttered all winter long into a lake...";
 	const int start = !strcmp(ptr, "start");
+	const char tmpfilecontent[] = "file_freeze_test_file";
 	char *cp;
 	char charbuf[128];
+	char *tmpfilename = NULL;
+	int fd = -1;
 	int r;
-	size_t orig_length;
+	size_t orig_length, len;
 	struct evbuffer_iovec v[1];
+	evutil_socket_t pair[2] = {-1, -1};
 
 	if (!start)
 		tt_str_op(ptr, ==, "end");
 
 	buf = evbuffer_new();
+	buf_two = evbuffer_new();
 	tmp_buf = evbuffer_new();
 	tt_assert(tmp_buf);
 
 	evbuffer_add(buf, string, strlen(string));
+	evbuffer_add(buf_two, "abc", 3);
+	evbuffer_add(tmp_buf, "xyz", 3);
 	evbuffer_freeze(buf, start); /* Freeze the start or the end.*/
+	evbuffer_freeze(buf_two, start);
+
+	/* create a socket pair for read/write test */
+	if (evutil_socketpair(AF_UNIX, SOCK_STREAM, 0, pair) == -1)
+		tt_abort_msg("socketpair failed");
+
+	evutil_make_socket_nonblocking(pair[0]);
+	evutil_make_socket_nonblocking(pair[1]);
```

Comment:
We have `TT_NEED_SOCKETPAIR`, but this requires `basic_setup`, and I don't see the reason why there is `nil_setup` used, can you convert it to `TT_NEED_SOCKETPAIR`? (this will avoid extra calls)

Reply:
hm, everything looks good, but requires some changes, smth like this:

```patch
diff --git a/test/regress_buffer.c b/test/regress_buffer.c
index b187a136..ab40f984 100644
--- a/test/regress_buffer.c
+++ b/test/regress_buffer.c
@@ -2494,13 +2494,14 @@ end:
 static void
 test_evbuffer_freeze(void *ptr)
 {
+       struct basic_test_data *testdata = ptr;
        struct evbuffer *buf = NULL, *buf_two = NULL, *tmp_buf = NULL;
        const char string[] = /* Year's End, Richard Wilbur */
            "I've known the wind by water banks to shake\n"
            "The late leaves down, which frozen where they fell\n"
            "And held in ice as dancers in a spell\n"
            "Fluttered all winter long into a lake...";
-       const int start = !strcmp(ptr, "start");
+       const int start = !strcmp(testdata->setup_data, "start");
        const char tmpfilecontent[] = "file_freeze_test_file";
        char *cp;
        char charbuf[128];
@@ -2512,7 +2513,7 @@ test_evbuffer_freeze(void *ptr)
        evutil_socket_t pair[2] = {-1, -1};

        if (!start)
-               tt_str_op(ptr, ==, "end");
+               tt_str_op(testdata->setup_data, ==, "end");

        buf = evbuffer_new();
        buf_two = evbuffer_new();
@@ -2811,8 +2812,8 @@ struct testcase_t evbuffer_testcases[] = {
        { "empty_reference_prepend_buffer", test_evbuffer_empty_reference_prepend_buffer, TT_FORK, NULL, NULL },
        { "peek", test_evbuffer_peek, 0, NULL, NULL },
        { "peek_first_gt", test_evbuffer_peek_first_gt, 0, NULL, NULL },
-       { "freeze_start", test_evbuffer_freeze, 0, &nil_setup, (void*)"start" },
-       { "freeze_end", test_evbuffer_freeze, 0, &nil_setup, (void*)"end" },
+       { "freeze_start", test_evbuffer_freeze, 0, &basic_setup, (void*)"start" },
+       { "freeze_end", test_evbuffer_freeze, 0, &basic_setup, (void*)"end" },
        { "add_iovec", test_evbuffer_add_iovec, 0, NULL, NULL},
        { "copyout", test_evbuffer_copyout, 0, NULL, NULL},
        { "file_segment_add_cleanup_cb", test_evbuffer_file_segment_add_cleanup_cb, 0, NULL, NULL },
```

Reply:
Thanks, I modified the file based on your reminder, please review.